apps/adelaidemetro/adelaide_metro.star

```diff
@@ -0,0 +1,1112 @@
+"""
+Applet: Adelaide Metro
+Summary: Adelaide Transit
+Description: Displays upcoming services for train stations and bus & tram stops around Adelaide.
+Author: M0ntyP
+
+Inspired by all the other great transit apps out there, I made one for my home town. I'd be surprised if anyone actually uses it :)
+"""
+
+load("render.star", "render")
+load("http.star", "http")
+load("humanize.star", "humanize")
+load("schema.star", "schema")
+load("encoding/json.star", "json")
+load("encoding/base64.star", "base64")
+load("cache.star", "cache")
+
+NEXTSCHED1_URL = "https://api-cloudfront.adelaidemetro.com.au/stops/next-scheduled-services?stop="
+STOPINFO_URL = "https://api-cloudfront.adelaidemetro.com.au/stops/info?stop="
+
+CACHE_TTL_SECS = 60
+
+def main(config):
+    SelectedStation = config.get("StationList", "16490")
+    TrainToCity = config.bool("TrainToCity", True)
+    TrainOrTramOrBus = config.get("TrainOrTramOrBus", "Bus")
+
+    if TrainOrTramOrBus == "Tram":
+        SelectedStation = config.get("TramStationList", "17753")
+
+    if TrainOrTramOrBus == "Bus":
+        SelectedStation = config.get("BusStop", 12471)
+
+    if TrainOrTramOrBus == "Train":
+        SelectedStation = config.get("StationList", "16490")
+
+    if TrainToCity == False:
+        SelectedStation = AwayStops(SelectedStation)
+
+    STOP_ID = str(SelectedStation)
+
+    NEXTSCHED_URL = NEXTSCHED1_URL + STOP_ID
+
+    # Cache the next service times for 1 min
+    NextSchedCacheData = get_cachable_data(NEXTSCHED_URL, 60)
+    NEXTSCHED_JSON = json.decode(NextSchedCacheData)
+
+    INFO_URL = STOPINFO_URL + STOP_ID
+    INFO_JSON = http.get(INFO_URL).json()
```

Comment:
Some caching here would be great!

Reply:
Hi @matslina , thanks for the feedback. I didn't want to cache that call as its the initial call to check if the Stop ID entered by the user is valid. So if they do enter an incorrect ID it wouldn't be cached, they get the "Invalid Stop ID" message and they can try again straight away. If the ID is correct then the data does get cached a few lines later at 62 & 63